src/services/public_http_server/handlers/relay_webhook/handlers/notify_delete.rs

```diff
@@ -77,115 +85,135 @@ pub async fn handle(msg: RelayIncomingMessage, state: &AppState) -> Result<(), R
 
     let sym_key =
         decode_key(&subscriber.sym_key).map_err(RelayMessageServerError::NotifyServerError)?; // TODO change to client error?
+    if msg.topic != topic_from_key(&sym_key) {
+        return Err(RelayMessageServerError::NotifyServerError(
+            NotifyServerError::TopicDoesNotMatchKey,
+        ))?; // TODO change to client error?
+    }
 
-    let msg = decrypt_message::<NotifyDelete, _>(envelope, &sym_key)
+    let req = decrypt_message::<NotifyDelete, _>(envelope, &sym_key)
         .map_err(RelayMessageServerError::NotifyServerError)?; // TODO change to client error?
-    info!("msg.id: {}", msg.id);
-    info!("msg.jsonrpc: {}", msg.jsonrpc); // TODO verify this
-    info!("msg.method: {}", msg.method); // TODO verify this
 
-    let request_auth = from_jwt::<SubscriptionDeleteRequestAuth>(&msg.params.delete_auth)
-        .map_err(RelayMessageClientError::JwtError)?;
-    info!(
-        "request_auth.shared_claims.iss: {:?}",
-        request_auth.shared_claims.iss
-    );
-    let request_iss_client_id = DecodedClientId::try_from_did_key(&request_auth.shared_claims.iss)
-        .map_err(AuthError::JwtIssNotDidKey)
-        .map_err(|e| RelayMessageServerError::NotifyServerError(e.into()))?; // TODO change to client error?
+    async fn handle(
+        state: &AppState,
+        msg: &RelayIncomingMessage,
+        req: &JsonRpcRequest<NotifyDelete>,
+        subscriber: &SubscriberWithScope,
+        project: &Project,
+        project_client_id: DecodedClientId,
+    ) -> Result<
+        (
+            ResponseAuth,
+            Vec<(SubscriptionWatcherQuery, Vec<NotifyServerSubscription>)>,
+        ),
+        RelayMessageError,
+    > {
+        info!("req.id: {}", req.id);
+        info!("req.jsonrpc: {}", req.jsonrpc); // TODO verify this
```

Comment:
Can you elaborate on what exactly needs to verify in these two lines?

Reply:
Just that `jsonrpc` matches `2.0` and the `method` matches the endpoint